### PR TITLE
[MOB-3546] flatlist refactor

### PR DESCRIPTION
**Pull request by roninopf**
## 🔹 JIRA Ticket(s) if any

* [MOB-3546](https://iterable.atlassian.net/browse/MOB-3546)

## ✏️ Description

* converts the previous `ScrollView` to a `FlatList` to keep the current functionality as well as lay the groundwork for session tracking (which will officially be implemented in MOB-3383)
* slight refactors to keep code consistent in style

### CODE REVIEW

File: ts/IterableInbox.tsx [hunk 1]
```diff
@@ -208,6 +208,15 @@ const IterableInbox = ({
...
 
+   function updateCurrentVisibleRows() {
+      // inboxDataModel.updateVisibleRows(getCurrentVisibleRows())
+   }
+
+   function getCurrentVisibleRows(): InboxImpressionRowInfo[] {
+
+      return []
+   }
+
```
Comment (evangreer91) [inline ts/IterableInbox.tsx:219]: This is for the implementation for the silent push correct?

Comment (roninopf) [reply ts/IterableInbox.tsx:219]: No, it's part of the upcoming session tracking

File: ts/IterableInboxMessageList.tsx [hunk 2]
```diff
@@ -30,31 +31,45 @@ const IterableInboxMessageList = ({
...
+
+   return (
+      <FlatList
+         scrollEnabled = {!swiping}
+         data = {rowViewModels}
+         renderItem = {({item, index}: {item: InboxRowViewModel, index: number }) => renderRowViewModel(item, index, index === rowViewModels.length - 1)}
+         keyExtractor = {(item: InboxRowViewModel) => item.inAppMessage.messageId}
+         viewabilityConfig = {inboxSessionViewabilityConfig}
+         onViewableItemsChanged = {inboxSessionItemsChanged}
+      />
```
Comment (evangreer91) [inline ts/IterableInboxMessageList.tsx:72]: I like the refactor. I think render Item should be written as follows:
renderItem = {(item: InboxRowViewModel, index: number) => renderRowViewModel(item, index, index === rowViewModels.length - 1)}

Comment (evangreer91) [reply ts/IterableInboxMessageList.tsx:72]: I see this is the required way to add a type for item and index

**Pull request merged**